services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/diagnostics/recommendationsTable.tsx

```diff
@@ -0,0 +1,183 @@
+    if (isSelected) {
+      newSelections[selectionIndex].students = newSelections[selectionIndex].students.filter(s => s !== student.id)
+    } else {
+      newSelections[selectionIndex].students.push(student.id);
+    }
+    setSelections(newSelections)
+  }
+  let checkbox = <span aria-label="Unchecked checkbox" className="quill-checkbox unselected" />
+  const assigned = isAssigned && <div className="assigned">{correctImage}<span>Assigned</span></div>
+  if (isSelected) {
+    checkbox = (<span aria-label="Checked checkbox" className="quill-checkbox selected" >
+      <img alt={smallWhiteCheckIcon.alt} src={smallWhiteCheckIcon.src} />
+    </span>)
+  }
+  return (<td className="recommendation-cell">
+    <button className={`interactive-wrapper ${isRecommended && isSelected && !isAssigned ? 'recommended-and-selected' : ''}`} onClick={isAssigned ? () => {} : toggleSelection} type="button">
+      {isRecommended ? asteriskIcon : <span />}
+      {assigned || checkbox}
+      <span />
+    </button>
+  </td>)
+}
+
+const StudentRow = ({ student, selections, recommendations, previouslyAssignedRecommendations, setSelections, }) => {
+  const { name, completed, id, } = student
+  const diagnosticNotCompletedMessage = completed ? null : <span className="diagnostic-not-completed">Diagnostic not completed</span>
+  const firstCell = (<th className="name-cell">
+    <div>
+      <StudentNameOrTooltip name={name} />
+      {diagnosticNotCompletedMessage}
+    </div>
+  </th>)
+
+  let selectionCells = selections.map((selection: Recommendation) => (<td key={selection.name} />))
+  if (completed) {
+    selectionCells = selections.map((selection: Recommendation, i: number) => {
+      const isAssigned = previouslyAssignedRecommendations.find(r => r.activity_pack_id === selection.activity_pack_id).students.includes(id)
+      const isRecommended = recommendations.find(r => r.activity_pack_id === selection.activity_pack_id).students.includes(id)
+      const isSelected = selection.students.includes(id)
+      return (<RecommendationCell
+        isAssigned={isAssigned}
+        isRecommended={isRecommended}
+        isSelected={isSelected}
+        key={`${id}-${selection.activity_pack_id}`}
+        selectionIndex={i}
+        selections={selections}
+        setSelections={setSelections}
+        student={student}
+      />)
+    })
+  }
+  return <tr key={name}>{firstCell}{selectionCells}</tr>
+}
+
+const RecommendationsTable = ({ recommendations, students, selections, previouslyAssignedRecommendations, setSelections, }: RecommendationsTableProps) => {
+  const [isSticky, setIsSticky] = React.useState(false);
+  const tableRef = React.useRef(null);
+  const [stickyTableStyle, setStickyTableStyle] = React.useState({
+    position: "fixed",
+    top: 0,
+    left: 0,
+    right: 0,
+    zIndex: 3
+  })
+
+  const handleScroll = React.useCallback(({ top, bottom, left, right, }) => {
+    if (top <= 0 && bottom > 92) {
```

Comment:
I think a comment explaining why we're using these values would be nice, like referencing the name of the class for a container just so we can know to make changes in both places should the need arise